lib/base/application.cpp

```diff
@@ -378,7 +378,7 @@ static void ReloadProcessCallback(const ProcessResult& pr)
 {
 	l_Restarting = false;
 
-	std::thread t(std::bind(&ReloadProcessCallbackInternal, pr));
+	std::thread t([&pr]() { ReloadProcessCallbackInternal(pr); });
```

Comment:
Explain with your own words what’s the problem with capturing `pr` by reference.

Reply:
I don't see any problems here, because the parameter is used only once in this function.

Reply:
Are you sure? Have a look what this constructor of std::thread does.

Reply:
I think by reference might be a problem here because ``thread`` captures also by const Ref. and that could end fatally if something goes wrong in the lambda.

Reply:
No, because it immediately starts a new thread which holds a reference to `pr` which may or may not exist anymore during usage due to stack unwinding.